### PR TITLE
Local planner and frontier explorer

### PR DESCRIPTION
**Pull request by ashwin2802**

### CODE REVIEW

File: frontier_explorer/include/frontier_explorer/goal_selector.hpp [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+        }
+
+        ros::NodeHandle nh_;
+        ros::NodeHandle nh_private_;
+
+        ros::Subscriber odom_sub_;
+
+        double clear_radius_;
+        double min_size_;
+        geometry_msgs::Pose curr_pose_;
```
Comment (gajena) [inline frontier_explorer/include/frontier_explorer/goal_selector.hpp]: change to mav_pose or something ... more intuitive

File: frontier_explorer/src/goal_selector.cpp [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+#include <visualization_msgs/MarkerArray.h>
+#include <visualization_msgs/Marker.h>
+
+namespace ariitk::frontier_explorer {
+
+FrontierComparator::FrontierComparator(const ros::NodeHandle& nh, const ros::NodeHandle& nh_private)
+    : nh_(nh)
+    , nh_private_(nh_private) {
+    nh_private_.getParam("min_frontier_size", min_size_);
+    nh_private_.getParam("clear_radius",      clear_radius_);
```
Comment (gajena) [inline frontier_explorer/src/goal_selector.cpp]: extra space after comma

Comment (gajena) [reply frontier_explorer/src/goal_selector.cpp]: check the complete PR again for these kind of things

Comment (ashwin2802) [reply frontier_explorer/src/goal_selector.cpp]: that was to actually align the param names. if that's not advised i'll edit it

File: frontier_explorer/CMakeLists.txt [hunk 3]
```diff
@@ -14,16 +14,17 @@ include_directories(${Eigen3_INCLUDE_DIRS})
...
+  src/goal_selector.cpp
 )
 target_link_libraries(${PROJECT_NAME} ${Eigen3_LIBRARIES})
 
 ############
 # BINARIES #
 ############
-cs_add_executable(frontier_evaluator_node
-  src/frontier_evaluator_node.cpp
+cs_add_executable(frontier_explorer_node
```
Comment (gajena) [inline frontier_explorer/CMakeLists.txt]: ${PROJECT_NAME}_node

File: frontier_explorer/include/frontier_explorer/goal_selector.hpp [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+        ros::Subscriber odom_sub_;
+
+        double clear_radius_;
+        double min_size_;
+        geometry_msgs::Pose mav_pose_;
+};
+
+class GoalSelector {
+    public:
+        GoalSelector(ros::NodeHandle& nh, ros::NodeHandle& nh_private);
```
Comment (gajena) [inline frontier_explorer/include/frontier_explorer/goal_selector.hpp]: check for const args in complete PR

File: frontier_explorer/src/goal_selector.cpp [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+            active_frontiers_.push_back(frontier);
+        }
+    }
+}
+
+void GoalSelector::visualizeActiveFrontiers() {
+    // cache active frontiers for viewing
+    static std::vector<ariitk_planning_msgs::Frontier> active_frontiers;
+
+    if (!active_frontiers_.empty())
```
Comment (gajena) [inline frontier_explorer/src/goal_selector.cpp:60]: scope braces

**Review by gajena — APPROVED** [review]

**Pull request merged**